server/src/main/java/org/elasticsearch/common/logging/DeprecationLogger.java

```diff
@@ -318,7 +320,10 @@ void deprecated(final Set<ThreadContext> threadContexts, final String message, f
         }
 
         if (log) {
-            logger.warn(message, params);
+            AccessController.doPrivileged((PrivilegedAction<Void>)() -> {
```

Comment:
Do we want this to be wrapped in a doPrivileged block for all calls or just for ones made from scripts?

Reply:
I think all callers is fine. It is essentially a noop for other callers, since they already run with at least the permissions of the server jar. Scripts are the only place with reduced privileges.